src/main/scala/de/heikoseeberger/sbtheader/HeaderCreator.scala

```diff
@@ -70,14 +72,21 @@ final class HeaderCreator private (fileType: FileType,
       case _       => "\n"
     }
 
-  private val newHeaderText = headerText.replace(headerNewLine, fileNewLine)
+  private val newHeaderText         = headerText.replace(headerNewLine, fileNewLine)
+  private val headerTextWithNewLine = headerText + newLine // the regex always checks for a new line at the end of a header text
 
   private val modifiedText =
     text match {
-      case headerPattern(`headerText`, _) => None
-      case headerPattern(_, body)         => Some(firstLine + newHeaderText + body)
-      case body if body.isEmpty           => None
-      case body                           => Some(firstLine + newHeaderText + body.replaceAll("""^\s+""", "")) // Trim left
+      case headerPattern(`headerText`, _)            => None
+      case headerPattern(`headerTextWithNewLine`, _) => None
```

Comment:
I named it `hText` now because `text` is used already and I wanted it to be short so the pattern doesn't break line.